operator/src/config.rs

```diff
@@ -250,10 +235,261 @@ pub fn get_env_variables(selector: &SparkNodeSelector) -> HashMap<String, String
 /// * `map` - Map containing option_name:option_value pairs
 /// * `assignment` - Used character to assign option_value to option_name (e.g. "=", " ", ":" ...)
 ///
-pub fn convert_map_to_string(map: &HashMap<String, String>, assignment: &str) -> String {
+fn convert_map_to_string(map: &HashMap<String, String>, assignment: &str) -> String {
     let mut data = String::new();
     for (key, value) in map {
         data.push_str(format!("{}{}{}\n", key, assignment, value).as_str());
     }
     data
 }
+
+/// All config map names follow a simple pattern: <name of SparkCluster object>-<NodeType name>-<SelectorHash>-cm
+/// That means multiple pods of one selector share one and the same config map
+///
+/// # Arguments
+/// * `cluster_name` - Current cluster name
+/// * `node_type` - SparkNodeType (master/worker/history-server)
+/// * `hash` - NodeSelector hash
+///
+pub fn create_config_map_name(cluster_name: &str, node_type: &SparkNodeType, hash: &str) -> String {
+    format!("{}-{}-{}-cm", cluster_name, node_type.as_str(), hash)
+}
+
+/// Create all required config maps and respective config map data.
+///
+/// # Arguments
+/// * `resource` - SparkCluster
+/// * `spec` - SparkClusterSpec containing common cluster config options
+/// * `selector` - SparkClusterSelector containing node specific config options
+/// * `node_type` - SparkNodeType (master/worker/history-server)
+/// * `hash` - NodeSelector hash
+///
+pub fn create_config_maps(
+    resource: &SparkCluster,
+    spec: &SparkClusterSpec,
+    selector: &SparkNodeSelector,
+    node_type: &SparkNodeType,
+    hash: &str,
+) -> OperatorResult<Vec<ConfigMap>> {
```

Comment:
Why a Vec when you return only one?

Reply:
In case we require multiple config maps in the future. For now it is overhead that is right.